src/tokens/ERC1155.vy

```diff
@@ -111,65 +180,88 @@ def supportsInterface(interface_id: bytes4) -> bool:
 def safeTransferFrom(owner: address, to: address, id: uint256, amount: uint256, data: Bytes[1024]):
     """
     @dev Transfers `amount` tokens of token type `id` from
-         `owner` to `to`.
+        `owner` to `to`.
     @notice Note that `to` cannot be the zero address. Also,
-            if the caller is not `owner`, it must have been
-            approved to spend `owner`'s tokens via `setApprovalForAll`.
-            Furthermore, `owner` must have a balance of tokens
-            of type `id` of at least `amount`. Eventually, if
-            `to` refers to a smart contract, it must implement
-            {IERC1155Receiver-onERC1155Received} and return the
-            acceptance magic value.
+        if the caller is not `owner`, it must have been
+        approved to spend `owner`'s tokens via `setApprovalForAll`.
+        Furthermore, `owner` must have a balance of tokens
+        of type `id` of at least `amount`. Eventually, if
+        `to` refers to a smart contract, it must implement
+        {IERC1155Receiver-onERC1155Received} and return the
+        acceptance magic value.
     @param owner The 20-byte address which previously
-           owned the token.
+        owned the token.
     @param to The 20-byte receiver address.
     @param id The 32-byte identifier of the token.
     @param amount The 32-byte token amount that is
-           being transferred.
+        being transferred.
     @param data The maximum 1024-byte additional data
-           with no specified format.
+        with no specified format.
     """
-    pass
+    assert owner == msg.sender or self.isApprovedForAll[owner][msg.sender], "ERC1155: Not authorized to transfer"
+    assert to != empty(address), "ERC1155: transfer to the zero address"
```

Comment:
everything that comes here could be put into an internal function `_safe_transfer_from`. The reason is composability (maybe some projects will import this function later via the `modules` possibility since they want to use it together with the `safeTransferFrom` function that however adds further custom checks (some special checks on the ids etc)) & consistency with the existing codebase.